oidc-controller/api/db/session.py

```diff
@@ -23,6 +28,32 @@ async def init_db():
     auth_session.create_index([("pres_exch_id", ASCENDING)], unique=True)
     auth_session.create_index([("pyop_auth_code", ASCENDING)], unique=True)
 
+    with open(
+        (
+            Path(__file__).parent.parent / "authSessions" / "sessiontimeout.json"
+        ).resolve()
+    ) as user_file:
+        experation_times: dict[str, int] = json.loads(user_file.read())
+        auth_session_states: list[str] = [str(i) for i in list(AuthSessionState)]
+        try:
+            for k, v in experation_times.items():
+                assert isinstance(k, str)
+                assert k in auth_session_states
+                auth_session.create_index(
+                    [("created_at", ASCENDING)],
+                    expireAfterSeconds=v + settings.CONTROLLER_PRESENTATION_BUFFER_TIME,
+                    name=k + "_ttl",
+                    partialFilterExpression={"proof_status": {"$eq": k}},
+                )
```

Comment:
Some thoughts on this (might change depending on the JSON file config pattern, TBD).
We're creating indexes in a loop and one per expiry condition, so we get multiple indexes:
![image](https://github.com/bcgov/vc-authn-oidc/assets/17445138/4623f3c6-2406-4c30-b8fc-f612ec175bf6)

Is the [partialFilterExpression](https://www.mongodb.com/docs/manual/core/index-partial/#create-a-partial-index) able to just take an OR condition instead? So there would be 1 index. Unless there's good reason to have multiple partial indexes.

If it **were** to just simplify to deleting all records regardless of state after a time it wouldn't have to be a partial index.

Also if using this pattern if one
- Applies the indexes to delete based on some config
- Changes the config
- Restarts the app
Will it then skip over changing the indexes? Because they are already there and it's only running `create_index` I don't _think_ it would remove previously created partials, or alter them?

Reply:
Previously we couldn't use OR but now that we are on mongo 7 we actually can which is great news! I should have just made this dependant on the mongo update. I'll switch this to an or now. As of now it was a constant once the index is set it would need to be removed however I can add support for this while moving to the new use of `OR`. Thanks for mentioning this!